model_signing/model.py

```diff
@@ -12,150 +13,64 @@
-    def verify(self, inputfn: Path, signaturefn: Path,
-               ignorepaths: [Path], offline: bool) -> VerificationResult:
-        try:
-            bundle_bytes = signaturefn.read_bytes()
-            bundle = Bundle.from_json(bundle_bytes)
-
-            contentio = io.BytesIO(Serializer.serialize_v1(
-                inputfn, chunk_size(), signaturefn, ignorepaths))
-            policy_ = policy.Identity(
-                identity=self.identity,
-                issuer=self.oidc_provider,
-            )
-            result = self.verifier.verify_artifact(input_=contentio,
-                                                   bundle=bundle,
-                                                   policy=policy_)
-            if result is None:
-                return VerificationResult()
-            return VerificationResult(success=False, reason=result.reason)
-        except Exception as e:
-            return VerificationResult(success=False,
-                                      reason=f"exception caught: {str(e)}")
-        raise ValueError("unreachable")
+from typing import Callable, TypeAlias
+import pathlib
+
+from model_signing.manifest import manifest
+from model_signing.signing import signing
+from model_signing.signature import verifying
+from model_signing.serialization import serialization
+
+PayloadGeneratorFunc: TypeAlias = \
+      Callable[[manifest.Manifest], signing.SigningPayload]
+
+
+def sign(
+    model_path: pathlib.Path,
+    signer: signing.Signer,
+    payload_generator: PayloadGeneratorFunc,
+    serializer: serialization.Serializer,
+    ignore_paths: list[pathlib.Path] = [],
+) -> signing.Signature:
+    """Provides a wrapper function for the steps necessary to sign a model.
+
+    Args:
+        model_path: the model to be signed
+        signer: the signer to be used
+        payload_generator: funtion to generate the
+            manifest
+        serializer: the serializer to be used for
+            the model
+        ignore_paths: paths that should be ignored during serialization Defaults to [].
+
+    Returns:
+        The model's signature.
+    """
+    manifest = serializer.serialize(model_path, ignore_paths=ignore_paths)
+    payload = payload_generator(manifest)
+    sig = signer.sign(payload)
+    return sig
+
+
+def verify(sig: signing.Signature,
+           verifier: signing.Verifier,
+           model_path: pathlib.Path,
+           serializer: serialization.Serializer,
+           ignore_paths: list[pathlib.Path] = []):
+    """verify is a simple wrapper to verify models
+
+    Args:
+        sig: the signature to be verified
+        verifier: the verifier
+        model_path: the path to the model to compare manifests
+        serializer: the serializer used to generate the local manifest
+        ignore_paths: paths that should be ignored during serialization. Defaults to [].
+
+    Raises:
+        verifying.VerificationError: on any verification error.
+    """
```

Comment:
This part did not change. I did not mention on every place where the indent or the format or the docstrings contents need to be updated